ibm/service/cos/resource_ibm_cos_bucket_object.go

```diff
@@ -124,7 +125,24 @@ func ResourceIBMCOSBucketObject() *schema.Resource {
 			"object_sql_url": {
 				Type:        schema.TypeString,
 				Computed:    true,
-				Description: "Access the object using an SQL Query instance.The reference url is used to perform queries against objects storing structured data.",
+				Description: "Access the object using an SQL Query instance. The reference url is used to perform queries against objects storing structured data.",
+			},
+			"object_lock_legal_hold_status": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(s3.ObjectLockLegalHoldStatus_Values(), false),
+				Description:  "An object lock configuration on the object, the valid states are ON/OFF. When ON prevents deletion of the object version.",
+			},
+			"object_lock_mode": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Retention modes apply different levels of protection to the objects.",
+			},
+			"object_lock_retain_until_date": {
+				Type:         schema.TypeString,
```

Comment:
object_lock_mode and object_lock_retain_until_date need to be used together always?

Reply:
Yes if any one of them is missing then it throws error as "malformed xml"

Reply:
added a required with for both them.

Reply:
addressed the comment